pysdsl/test/test_sorted_int_stack.py

```diff
@@ -0,0 +1,52 @@
+from pysdsl import SortedIntStack as stack
+import random
+
+def test_push_pop_top_empty():
+    s = stack(max_value=1000)
+    a = []
+
+    assert(s.empty() and s.size == 0 and len(s) == 0)
+
+    a = list(sorted(set([random.randint(0, 999) for _ in range(1000)])))
+    
+    for i in range(len(a)):
+        s.push(a[i])
+        assert(s.top() == a[i] and \
+               not s.empty() and \
+               len(s) == i + 1 and \
+               s.size == i + 1)
+
+    while len(a) > 0:
+        assert(len(a) == len(s) and \
+               len(a) == s.size and \
+               not s.empty() and \
+               a[-1] == s.top() and \
+               a[-1] == s.pop())
+        a.pop()
+
+    assert(s.empty() and s.size == 0 and len(s) == 0)
+
+
+def test_copy_assign():
+    s1 = stack(max_value=1000)
+    s2 = stack(max_value=2000)
+
+    s1.push(10); s1.push(100); s1.push(1000)
+    s2.push(20); s2.push(200); s2.push(2000)
+
+    # copy
+    s3 = stack(s1)
+
+    # reference
+    s4 = s2
+
+    assert(len(s1) == len(s3) and \
+           len(s2) == len(s4))
+
+    while len(s1) > 0:
+        assert(s1.pop() == s3.pop())
+
+    while len(s2) > 0:
+        assert(s2.top() == s4.top() and \
+               len(s2) == len(s4))
+        s2.pop()
```

Comment:
it might be a good idea to test what happens when we pop from empty stack or when too large value is pushed (exceptions is expected, not segfault)